lnwallet/chanfunding/canned_assembler.go

```diff
@@ -207,6 +233,7 @@ func (c *CannedAssembler) ProvisionChannel(req *Request) (Intent, error) {
 		remoteKey:  c.remoteKey,
 		chanPoint:  &c.chanPoint,
 		thawHeight: c.thawHeight,
+		musig2:     c.musig2,
```

Comment:
Good q...looks like an inconsistency: we want to make `NewCannedAssembler` up front, and that point we already know it needs to be musig2. The request is for the actual wallet assembler, since it needs to decide on the script format. 

Eg: take a look at the new `ProvisionChannel` method.